### PR TITLE
Quebec Basic Personal Amount

### LINKED ISSUE #396: Quebec Basic Personal Amount
[Reference](https://www.revenuquebec.ca/documents/en/formulaires/tp/2022-12/TP-1.D-V%282022-12%29.pdf)

### PR DESCRIPTION
**Pull request by Shinuing**
Fixes #396

### CODE REVIEW

File: policyengine_canada/variables/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+class qc_basic_personal_amount(Variable):
+    value_type = float
+    entity = Person
+    label = "Quebec basic personal amount"
+    unit = CAD
+    definition_period = YEAR
+    defined_for = ProvinceCode.QC
+    reference = "https://www.revenuquebec.ca/documents/en/formulaires/tp/2022-12/TP-1.D-V%282022-12%29.pdf"
+
+    adds = "gov.provinces.qc.tax.income.credits.basic_personal_amount"
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.py:13]: need to switch to formula that returns the parameter

File: policyengine_canada/variables/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+
+class qc_basic_personal_amount(Variable):
+    value_type = float
+    entity = Person
+    label = "Quebec basic personal amount"
+    unit = CAD
+    definition_period = YEAR
+    defined_for = ProvinceCode.QC
+    reference = "https://www.revenuquebec.ca/documents/en/formulaires/tp/2022-12/TP-1.D-V%282022-12%29.pdf"
```
Comment (Snownow0w) [inline policyengine_canada/variables/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.py:11]: Include all legal code:
    reference = (
"https://www.revenuquebec.ca/documents/en/formulaires/tp/2022-12/TP-1.D-V%282022-12%29.pdf", 
"https://www.revenuquebec.ca/en/citizens/income-tax-return/completing-your-income-tax-return/how-to-complete-your-income-tax-return/line-by-line-help/350-to-398-1-non-refundable-tax-credits/line-350/",
)

File: changelog_entry.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Quebec basic personal amount.
```
Comment (Snownow0w) [inline changelog_entry.yaml:4]: aa

File: policyengine_canada/variables/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    label = "Quebec basic personal amount"
+    unit = CAD
+    definition_period = YEAR
+    defined_for = ProvinceCode.QC
+    reference = (
+        "https://www.revenuquebec.ca/documents/en/formulaires/tp/2022-12/TP-1.D-V%282022-12%29.pdf",
+        "https://www.revenuquebec.ca/en/citizens/income-tax-return/completing-your-income-tax-return/how-to-complete-your-income-tax-return/line-by-line-help/350-to-398-1-non-refundable-tax-credits/line-350/",
+    )
+
+    def formula(household, period, parameters):
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.py:16]: need to check for is_head

Comment (Shinuing) [reply policyengine_canada/variables/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.py]: Also added a test file for `is_head`, let me know if no needed

File: policyengine_canada/tests/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+- name: Quebec basic personal amount for household head
+  period: 2022
+  input:
+    province_code: QC
+    is_head: True
```
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.yaml:5]:
```suggestion
    is_head: true
```

File: policyengine_canada/tests/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  output:
+    qc_basic_personal_amount: 16_143
+
+- name: Quebec basic personal amount for non-head of household individuals
+  period: 2022
+  input:
+    province_code: QC
+    is_head: False
```
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.yaml:13]:
```suggestion
    is_head: false
```

File: policyengine_canada/parameters/gov/provinces/qc/tax/income/credits/basic_personal_amount.yaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+description: Quebec provides the this Basic Personal Amount.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/qc/tax/income/credits/basic_personal_amount.yaml:1]:
```suggestion
description: Quebec provides this Basic Personal Amount.
```

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_canada/parameters/gov/provinces/qc/tax/income/credits/basic_personal_amount.yaml [hunk 8]
```diff
@@ -0,0 +1,13 @@
+description: Quebec provides this Basic Personal Amount.
+values:
+  2022-01-01: 16_143
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Quebec Basic Personal Amount
+  reference:
+    - title: Quebec 2022 Income Tax Return - Non-refundable tax credits
+      href: https://www.revenuquebec.ca/documents/en/formulaires/tp/2022-12/TP-1.D-V%282022-12%29.pdf
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/qc/tax/income/credits/basic_personal_amount.yaml:10]:
```suggestion
      href: https://www.revenuquebec.ca/documents/en/formulaires/tp/2022-12/TP-1.D-V%282022-12%29.pdf#page=3
```

File: policyengine_canada/parameters/gov/provinces/qc/tax/income/credits/basic_personal_amount.yaml [hunk 9]
```diff
@@ -0,0 +1,13 @@
+description: Quebec provides this Basic Personal Amount.
+values:
+  2022-01-01: 16_143
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Quebec Basic Personal Amount
+  reference:
+    - title: Quebec 2022 Income Tax Return - Non-refundable tax credits
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/qc/tax/income/credits/basic_personal_amount.yaml:9]:
```suggestion
    - title: Quebec 2022 Income Tax Return - Non-refundable tax credits - Line 350
```

File: policyengine_canada/variables/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    definition_period = YEAR
+    defined_for = ProvinceCode.QC
+    reference = (
+        "https://www.revenuquebec.ca/documents/en/formulaires/tp/2022-12/TP-1.D-V%282022-12%29.pdf",
+        "https://www.revenuquebec.ca/en/citizens/income-tax-return/completing-your-income-tax-return/how-to-complete-your-income-tax-return/line-by-line-help/350-to-398-1-non-refundable-tax-credits/line-350/",
+    )
+
+    def formula(person, period, parameters):
+        p = parameters(period).gov.provinces.qc.tax.income.credits
+        head = person("is_head", period)
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.py:18]: As with the other provinces I'm not clear how this works for spouses. Seems like a higher level question though around tax filing units in Canada. I see this form requests some information about spousal earnings.

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_canada/variables/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.py (hunk 10 above)
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.py:18]:
```suggestion
```

File: policyengine_canada/variables/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.py [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+        return head * p.basic_personal_amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/qc/tax/income/credits/qc_basic_personal_amount.py:19]:
```suggestion
        return p.basic_personal_amount
```

File: policyengine_canada/parameters/gov/provinces/qc/tax/income/credits/basic_personal_amount.yaml [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+  2022-01-01: 16_143
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Quebec Basic Personal Amount
+  reference:
+    - title: Quebec 2022 Income Tax Return - Non-refundable tax credits - Line 350
+      href: https://www.revenuquebec.ca/documents/en/formulaires/tp/2022-12/TP-1.D-V%282022-12%29.pdf#page=3
+    - title: Revenu Quebec – Basic Personal Amount (Line 350)
+      href: https://www.revenuquebec.ca/en/citizens/income-tax-return/completing-your-income-tax-return/how-to-complete-your-income-tax-return/line-by-line-help/350-to-398-1-non-refundable-tax-credits/line-350/
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/qc/tax/income/credits/basic_personal_amount.yaml:12]: is there a legal code (quebec income tax act f.e.) that we can cite as well - just to have some legal code

Comment (Shinuing) [reply policyengine_canada/parameters/gov/provinces/qc/tax/income/credits/basic_personal_amount.yaml:12]: Correct me if I'm wrong. I guess the Basic Personal Amount (BPA) is the same as the Personal Tax Credit?

https://www.legisquebec.gouv.qc.ca/en/document/cs/I-3?langCont=fr#se:752_0_0_1

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**